src/App.tsx

```diff
@@ -1,61 +1,109 @@
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+let maxId = [...todosFromServer].sort((a, b) => b.id - a.id)[0].id;
+
+const addTodo = (todo: string, userId: number) => {
+  maxId += 1;
+
+  const todoAdd = {
+    id: maxId,
+    completed: false,
+    title: todo,
+    userId,
+    user: getUser(userId),
+  };
+
+  todos.push(todoAdd);
+};
 
 export const App = () => {
+  const [toDo, setToDo] = useState('');
+  const [user, setUser] = useState('');
+  const [errorToDo, setErrorToDo] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (toDo.length > 0 && Number(user) !== 0) {
+      addTodo(toDo, Number(user));
+      setToDo('');
+      setUser('');
+      setErrorToDo(false);
+      setErrorUser(false);
+    } else if (toDo.length === 0) {
+      setErrorToDo(true);
+    } else if (Number(user) === 0) {
+      setErrorUser(true);
+    }
+  };
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <h1 className="App__title">Add todo for users</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label className="field__label">
+            <p className="form__title"> Title: </p>
+            <input
+              className="field__form"
+              type="text"
+              data-cy="titleInput"
+              value={toDo}
+              onChange={(event) => setToDo(event.target.value)}
+            />
+          </label>
+          {errorToDo && toDo.length < 1
+            && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <p className="form__title"> Users: </p>
+          <select
+            className="field__form"
+            data-cy="userSelect"
+            name="user"
+            value={user}
+            onChange={(event) => setUser(event.target.value)}
+          >
+            <option value="0">Choose a user</option>
+            {usersFromServer.map(({ id, name }) => (
+              <option value={id} key={id}>{name}</option>
+            ))}
           </select>
-
-          <span className="error">Please choose a user</span>
+          {errorUser && Number(user) === 0
```

Comment:
```suggestion
          {!errorUser && Number(user) === 0
```

Reply:
error should be shown after 'add' button was clicked.
<img width="525" alt="image" src="https://user-images.githubusercontent.com/106744410/184815757-5db94b2a-4058-4fd9-ac4c-44f7a5c010bc.png">